Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.cpp

```diff
@@ -59,8 +60,10 @@ class RemoteAudioMediaStreamTrackRendererInternalUnitManager::Unit
     void start(ConsumerSharedCARingBuffer::Handle&&, IPC::Semaphore&&);
     void stop();
     void setAudioOutputDevice(const String&);
-    OSStatus render(size_t sampleCount, AudioBufferList&, uint64_t sampleTime, double hostTime, AudioUnitRenderActionFlags&);
-    void notifyReset();
+
+    // WebCore::AudioMediaStreamTrackRendererInternal::Client
```

Comment:
Can these be private?

Reply:
Using `private` in this case means “called only polymorphically, not directly through a pointer specifically of this class’s type”. We can debate how valuable that is, but it’s not meaningless and to me it doesn’t come across as misleading about how public the function is in the base class. In fact whether the function being overridden in the base class might change over time and I would not want to update the visibility in all the derived classs. It’s a personal preference of mine, to clarify which functions are intended to be called in what fashion, not truly a project standard. I find it particularly helpful when a function is not helpful called non-polymorphically, like a function `isC` that returns `true` in a class named `C`. But also I think it helps make clear what a class is by cutting down on public members and it’s consistent with a general “everything as private as possible” pattern that can help make future refactoring simpler.

Reply:
Understood.
I moved the base class to private inheritance and moved the functions to private block.